From 3ce4ce9ba0c2c7e5e74c560aa9e394764e818be5 Mon Sep 17 00:00:00 2001
From: WingMan Kwok <w-kwok2@ti.com>
Date: Wed, 13 Jan 2016 08:20:51 -0500
Subject: [PATCH 04/47] ARM: dts: keystone: use syscon regmap to enable shared
 access of QMSS INTD

This patch adds syscon regmap entry for QMSS INTD (interrupt distributor)
and thus enabling shared access of the INTD by more than one PDSPs.

Signed-off-by: WingMan Kwok <w-kwok2@ti.com>
---
 arch/arm/boot/dts/keystone-k2e-netcp.dtsi  | 7 ++++++-
 arch/arm/boot/dts/keystone-k2hk-netcp.dtsi | 7 ++++++-
 arch/arm/boot/dts/keystone-k2l-netcp.dtsi  | 7 ++++++-
 3 files changed, 18 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/keystone-k2e-netcp.dtsi b/arch/arm/boot/dts/keystone-k2e-netcp.dtsi
index 9fd0b06..7ae7ca0 100644
--- a/arch/arm/boot/dts/keystone-k2e-netcp.dtsi
+++ b/arch/arm/boot/dts/keystone-k2e-netcp.dtsi
@@ -8,6 +8,11 @@
  * published by the Free Software Foundation.
  */
 
+pdsp_intd: intd@2a0c000 {
+	compatible	= "syscon";
+	reg		= <0x2a0c000 0x3c8>;
+};
+
 qmss: qmss@2a40000 {
 	compatible = "ti,keystone-navigator-qmss";
 	dma-coherent;
@@ -103,9 +108,9 @@ qmss: qmss@2a40000 {
 		#size-cells = <1>;
 		ranges;
 		pdsp0@0x2a10000 {
+			syscon-intd = <&pdsp_intd>;
 			reg = <0x2a10000 0x1000    /*iram */
 			       0x2a0f000 0x100     /*reg*/
-			       0x2a0c000 0x3c8	   /*intd */
 			       0x2a20000 0x4000>;  /*cmd*/
 			id = <0>;
 		};
diff --git a/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi b/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
index d9c9b35..67b84da 100644
--- a/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
+++ b/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
@@ -8,6 +8,11 @@
  * published by the Free Software Foundation.
  */
 
+pdsp_intd: intd@2a0c000 {
+	compatible	= "syscon";
+	reg		= <0x2a0c000 0x3c8>;
+};
+
 qmss: qmss@2a40000 {
 	compatible = "ti,keystone-navigator-qmss";
 	dma-coherent;
@@ -116,9 +121,9 @@ qmss: qmss@2a40000 {
 		#size-cells = <1>;
 		ranges;
 		pdsp0@0x2a10000 {
+			syscon-intd = <&pdsp_intd>;
 			reg = <0x2a10000 0x1000    /*iram */
 			       0x2a0f000 0x100     /*reg*/
-			       0x2a0c000 0x3c8	   /*intd */
 			       0x2a20000 0x4000>;  /*cmd*/
 			id = <0>;
 		};
diff --git a/arch/arm/boot/dts/keystone-k2l-netcp.dtsi b/arch/arm/boot/dts/keystone-k2l-netcp.dtsi
index 8a54126..6b3d552 100644
--- a/arch/arm/boot/dts/keystone-k2l-netcp.dtsi
+++ b/arch/arm/boot/dts/keystone-k2l-netcp.dtsi
@@ -8,6 +8,11 @@
  * published by the Free Software Foundation.
  */
 
+pdsp_intd: intd@2a0c000 {
+	compatible	= "syscon";
+	reg		= <0x2a0c000 0x3c8>;
+};
+
 qmss: qmss@2a40000 {
 	compatible = "ti,keystone-navigator-qmss";
 	dma-coherent;
@@ -98,9 +103,9 @@ qmss: qmss@2a40000 {
 		#size-cells = <1>;
 		ranges;
 		pdsp0@0x2a10000 {
+			syscon-intd = <&pdsp_intd>;
 			reg = <0x2a10000 0x1000    /*iram */
 			       0x2a0f000 0x100     /*reg*/
-			       0x2a0c000 0x3c8	   /*intd */
 			       0x2a20000 0x4000>;  /*cmd*/
 			id = <0>;
 		};
-- 
1.9.1

